### PR TITLE
Couple of corner case fixes in ellipseFromEllipticalArc

### PR DESCRIPTION
**Pull request by karanjitsingh**
For path data "M100,350 l 50,-25 a25,25 -30 0,1 50,-25 l 50,-25"
Line 2073: argument for Math.sqrt is negative
fix: pass through Math.abs

For path data "M100,350 l 50,-25 a25,75 -30 0,1 50,-25 l 50,-25"
Line 2009: ratio(u,v) is -1.0000000000000002 and hence Math.acos(ratio(u,v)) becomes NaN
fix: a funciton clamp(val, min, max) that clamps the value 'val' between the range [min,max]

"M100,350 l 50,-25 a25,25 -30 0,1 50,-25 l 50,-25"
firstfix

"M100,350 l 50,-25 a25,75 -30 0,1 50,-25 l 50,-25"
secondfix

Red stroke represents svg render
Black stroke represents Path2D render

### CODE REVIEW

**Comment by karanjitsingh** [comment]
I signed in!

File: canvas.js [hunk 1]
```diff
@@ -1996,13 +1996,17 @@ if (typeof Path2D !== 'function' ||
           function ratio(u, v) {
             return dot(u,v) / (mag(u)*mag(v))
           }
+
+          function clamp(value,min,max) {
```
Comment (jcgregorio) [inline canvas.js]: Super minor, but can you add spaces after the commas?

function clamp(value, min, max) {

Comment (karanjitsingh) [reply canvas.js]: Done!

**Comment by jcgregorio** [comment]
Thanks!

**Comment by Link2Twenty** [comment]
Also the clamp function has a 'value' argument but then references it as 'val' preventing the function from working.

**Pull request merged**